src/opm/output/eclipse/AggregateConnectionData.cpp

```diff
@@ -16,25 +16,22 @@
   You should have received a copy of the GNU General Public License
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
+#include <cassert>
+#include <cstddef>
+#include <iostream>
+#include <cmath>
 
-#include <opm/output/eclipse/AggregateConnectionData.hpp>
```

Comment:
I'd really prefer that the CPP file's header be the first `#include`.  It ensures that the header can be compiled in isolation (i.e., that it includes&mdash;or forward declares&mdash;all the types it needs in its Interface).  Also, for the same reason, standard headers really should be last.

Reply:
Personally I prefer the total opposite - i.e. the most general headers - like `#include <vector>` and  then ordered in increasing degree of relevance - so that the final `#include` corresponds to the actual CPP file - and then the actual code begins; flows like a charm!

> I'd really prefer that the CPP file's header be the first #include. It ensures that the header can be compiled in isolation

I see that you have a valid point; although I would like to point out that the way I prefer it is the main way to organize things in opm-common; and all-in-all that has worked.

Reply:
@bska : Please indicate if you are (reasonably ...) happy to see this merged after my arguments; I will revert the offending changes if you insist.

Reply:
To be honest, I'm a little on the fence.

I think I still would like the CPP file's header first&mdash;at least in `src/opm/output/eclipse/*.cpp`&mdash;to ensure header idempotency, but the other ordering changes I don't really care too much about one way or another.  If we do go through with the `#include` reordering, then we should take the opportunity to ensure that standard headers appear in lexicographically sorted order too.

For what it's worth our [suggested coding standard](https://wiki.opm-project.org/index.php?title=Suggested_coding_standard#Programming_guidelines) specifies that includes be ordered as

1. The header file for the current class or function

2. Other OPM headers

3. Third-party headers (Dune, Boost etc.)

4. System headers (`<algorithm>`, `<vector>` etc.)

In my own work I tend to put Boost headers after the standard headers so I admittedly don't follow that specification too closely.